library/common/common/lambda_logger_delegate.cc

```diff
@@ -12,7 +12,10 @@ LambdaDelegate::LambdaDelegate(envoy_logger logger, DelegatingLogSinkSharedPtr l
   setDelegate();
 }
 
-LambdaDelegate::~LambdaDelegate() { restoreDelegate(); }
+LambdaDelegate::~LambdaDelegate() {
+  restoreDelegate();
+  logger_.release(logger_.context);
```

Comment:
Excellent. Exactly the right place to put this.

Reply:
Yep, very clean this way.